server.py

```diff
@@ -141,68 +204,211 @@ def connect_client(self):
+            champion_select_result = self.champion_select(client_number)
 
-                if champion_select_result == CHAMPION_SELECT:
-                    continue
-                elif champion_select_result == LEFT_CHAMPION_SELECT:
-                    break
+            if champion_select_result == CHAMPION_SELECT:
+                continue
+            elif champion_select_result == LEFT_CHAMPION_SELECT:
+                break
 
-                while True:
-                    lobby_result = self.lobby(client_number)
-                    if lobby_result == CONNECTION_IS_DEACTIVATED:
-                        connection = CONNECTION_IS_DEACTIVATED
-                        break
-                    elif lobby_result == BACK_TO_CHAMPION_SELECT:
-                        break
+            while True:
+                lobby_result = self.lobby(client_number)
+                if lobby_result == CONNECTION_IS_DEACTIVATED:
+                    connection = CONNECTION_IS_DEACTIVATED
+                    break
+                elif lobby_result == BACK_TO_CHAMPION_SELECT:
+                    break
+                elif lobby_result == START_GAME:
+                    game_start = GAME_IS_STARTED
+                    break
 
-            except ConnectionResetError:
-                break
+            if game_start:
+                self.game(client_number)
+                connection = CONNECTION_IS_DEACTIVATED
 
         self.client_number[client_number] = FREE
         self.connections_needed_to_start += CLOSED_CONNECTION
         self.number_of_started_connections -= CLOSED_CONNECTION
         self.number_of_started_games -= GAME_IS_ENDED
         self.client_sockets[client_number].close()
+        self.chosen_champions[client_number] = CHAMPION_IS_NOT_CHOSEN
         sys.exit()
 
     def champion_select(self, client_number):
-        chosen_champion = pickle.loads(self.client_sockets[client_number].recv(BUFFER_SIZE))
+        try:
+            chosen_champion = pickle.loads(self.client_sockets[client_number].recv(BUFFER_SIZE))
 
-        if chosen_champion == CLIENT_QUIT:
-            return LEFT_CHAMPION_SELECT
+            if chosen_champion == CLIENT_QUIT:
+                return LEFT_CHAMPION_SELECT
 
-        if self.server_status == SERVER_IS_CLOSING:
-            self.client_sockets[client_number].sendall(SERVER_QUIT_SIGNAL_IN_CHAMPION_SELECT)
-            return LEFT_CHAMPION_SELECT
+            if self.server_status == SERVER_IS_CLOSING:
+                self.client_sockets[client_number].sendall(SERVER_QUIT_SIGNAL_IN_CHAMPION_SELECT)
+                return LEFT_CHAMPION_SELECT
 
-        if chosen_champion in self.chosen_champions:
-            self.client_sockets[client_number].sendall(CHAMPION_IS_NOT_AVAILABLE)
-            return CHAMPION_SELECT
+            if chosen_champion in self.chosen_champions:
+                self.client_sockets[client_number].sendall(CHAMPION_IS_NOT_AVAILABLE)
```

Comment:
aren't those if statements to deep? Like to far away from left border of the file?

Reply:
due to try